### PR TITLE
ISPN-5797 NullPointerException in ClientListenerRegistry under load

### PR DESCRIPTION
**Pull request by wburns**
* Made sure null value and metadata are supported

https://issues.jboss.org/browse/ISPN-5797

### CODE REVIEW

File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala [hunk 1]
```diff
@@ -183,8 +183,12 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
       @CacheEntryExpired
       def onCacheEvent(event: CacheEntryEvent[Bytes, Bytes]) {
          if (isSendEvent(event)) {
-            val dataVersion = event.getMetadata.version().asInstanceOf[NumericVersion].getVersion
-            sendEvent(event.getKey, event.getValue, dataVersion, event)
+            sendEvent(event.getKey, event.getValue, if (event.getMetadata == null) {
```
Comment (gustavonalle) [inline server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala]: Minor: the 3rd param to ```sendEvent``` could be written in a more functional way as:

```
Option(event.getMetadata)
      .map(_.version().asInstanceOf[NumericVersion].getVersion)
      .getOrElse(null.asInstanceOf[Long])
```

It'd be nicer without the ugly casts, but this cast is around in several places. It assumes ```NumericVersion``` is always used, so I wonder if there's a valid use case for setting SIMPLE versioning scheme on the server. If so, it'd break in several places. Not related to this PR anyway.

File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala [hunk 2]
```diff
@@ -183,8 +183,12 @@ class ClientListenerRegistry(configuration: HotRodServerConfiguration) extends L
...
+               null.asInstanceOf[Long]
+            } else {
+               event.getMetadata.version().asInstanceOf[NumericVersion].getVersion
+            }, event);
```
Comment (gustavonalle) [inline server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala]: No need for ```;```

File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala (hunk 1 above)
Comment (wburns) [inline server/hotrod/src/main/scala/org/infinispan/server/hotrod/ClientListenerRegistry.scala]: Cool, I will update it.   The issue is not with having versioning enabled or not.  The problem is that there are some cases where the metadata and value are not available from an expiration event and thus we have to handle those null values.

**Comment by wburns** [comment]
Updated per comments.

**Comment by wburns** [comment]
This needs to also be cherry-picked into 8.0.x branch

**Comment by gustavonalle** [comment]
pulling

**Comment by gustavonalle** [comment]
Integrated in master and 8.0.x, thanks @wburns !

**Pull request closed without merging**